src/vm/eventpipeeventsource.h

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef __EVENTPIPE_EVENTSOURCE_H__
+#define __EVENTPIPE_EVENTSOURCE_H__
+
+#ifdef FEATURE_PERFTRACING
+
+class EventPipeProvider;
+class EventPipeEvent;
+class EventPipeSession;
+
+class EventPipeEventSource
+{
+private:
+    const static WCHAR* s_pProviderName;
+    EventPipeProvider *m_pProvider;
+
+    const static WCHAR* s_pProcessInfoEventName;
+    EventPipeEvent *m_pProcessInfoEvent;
+
+public:
+    EventPipeEventSource();
+    ~EventPipeEventSource();
+
+    void Enable(EventPipeSession *pSession);
```

Comment:
This is enabling the provider for a specific session.  Sessions don't require explicit disable per provider - the session itself can be enabled/disabled.  There isn't a way to disable a provider within an enabled session (or even a disabled session).

Reply:
OK, cool. So once a provider is enabled in a session, it will always be a part of that session until the session dies?

Reply:
Yes, that's right.  @vancem had recommended changing this at some point so that could happen.  If it does, we'd probably still leave this as is because we want this to always be on (at least the LogAlways events) because they are things we believe should exist in all traces.